### PR TITLE
[EASY] Add log for volume based fee

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Haris noticed we are missing logs for volume based fees in the driver.

### CODE REVIEW

**Review by squadgazzz — APPROVED** [review]
Why do we want to log only Volume based fees?

File: crates/driver/src/domain/competition/solution/fee.rs [hunk 1]
```diff
@@ -126,7 +126,11 @@ impl Fulfillment {
                 )?;
                 self.calculate_fee(price_limits, prices, *factor, *max_volume_factor)
             }
-            FeePolicy::Volume { factor } => self.fee_from_volume(prices, *factor),
+            FeePolicy::Volume { factor } => {
+                let fee_from_volume = self.fee_from_volume(prices, *factor)?;
+                tracing::debug!(uid=?self.order().uid, ?fee_from_volume, executed=?self.executed(), surplus_fee=?self.surplus_fee(), "calculated protocol fee");
```
Comment (MartinquaXD) [inline crates/driver/src/domain/competition/solution/fee.rs:131]: `cargo fmt` is not always able to format long macros.
Could you manually add the required line breaks here and for the `price improvement` log?

**Review by MartinquaXD — APPROVED** [review]

**Comment by sunce86** [comment]
> Why do we want to log only Volume based fees?

Other types are already logged inside `calculate_fee` function.

File: crates/driver/src/domain/competition/solution/fee.rs (hunk 1 above)
Comment (sunce86) [reply crates/driver/src/domain/competition/solution/fee.rs:131]: cargo fmt returns errors when i manually add breaks

**Pull request merged**